corehq/apps/locations/bulk_management.py

```diff
@@ -883,27 +885,45 @@ def append_at_bottom(parent_type):
 
         return top_to_bottom_locations
 
-    delete_locations = []
-    with transaction.atomic():
-        for loc in order_by_location_type():
-            if loc.do_delete:
-                delete_locations.append(loc)
+    # _seen is intentionally a mutable cache of all ids seen so far
```

Comment:
How would you feel about instead doing
```python
_seen = set()
def iter_unprocessed_ancestor_ids(stubs):
    ...
```
That way seems like the lifecycle and scope of `_seen` is clearer

Reply:
hehe, sure I'll change it. I like this way because it makes it clear that `_seen` is only referenced within the scope of `iter_unprocessed_ancestor_ids`.